main.cpp

```diff
@@ -51,22 +52,401 @@ If you need inspiration for what to write, take a look at previously approved st
+{
+    int x = mastHeight - distanceToWater;
+    std::cout << "Heel distance is: " << x << std::endl;
+    return x;
+}
 
 /*
  UDT 2:
  */
+struct SchoolDay
+{   
+    SchoolDay();
+    ~SchoolDay();
+    std::string dayOfTheWeek;
+    float length;
+    int numberOfKids = 15;
+    bool runningLate;
+    bool halfDay = false;
+    bool isRaining = false;
+    void splitUpClass( int numOfGroups );
+    void shortenSchoolDay( float lengthDecrease );
+    void sunCameOut();
+};
+SchoolDay::SchoolDay()
+{ 
+    dayOfTheWeek = "Monday";
+    length = 6.f;
+    runningLate = false;
+    std::cout << "A schoolday was created" << std::endl;
+
+}
+SchoolDay::~SchoolDay()
+{
+    std::cout << "A schoolday was destroyed" << std::endl;
+}
+
+void SchoolDay::splitUpClass(int numOfGroups)
+{   
+    if (numberOfKids != 0)
+    {
+        numberOfKids /= numOfGroups;
+    }
+    else
+    {
+        std::cout << "there are" << numberOfKids << "kids in class!" << std::endl;
+    }
+}
+void SchoolDay::shortenSchoolDay( float lengthDecrease )
+{
+    if (halfDay == false)
+    {
+        length -= lengthDecrease;
+        std::cout << "SchoolDay length is now: " << length << std::endl;
+        halfDay = true;
+    }
+    
+}   
+void SchoolDay::sunCameOut()
+{
+    if (isRaining == true)
+    {
+        isRaining = false;
+        std::cout << "sun came out" << std::endl;
+    }
+    else
+    {
+        std::cout << "sun is already out" << std::endl;
+    }
+}
 
 /*
  UDT 3:
  */
+struct SurfReport
+{
+    SurfReport();
+    ~SurfReport();
+    float waveHeight;
+    float wavePeriod;
+    std::string windDirection;
+    int windSpeed;
+    bool isRaining = false;
+    bool isBusy = false;
+    struct Surfer
+    {
+        Surfer();
+        ~Surfer();
+        bool atBeach = true;
+        int surferAge;
+        int surferSkill = 0;
+        float boardSize = 6.8f;
+        bool havingLesson = false;
+
+        void levelUp( int improvementAmount );
+        void getTubed( int currentStokeLevel );
+        void wipeOut( int currentStokeLevel );
+    };
+
+    void increaseWaveHeight( float newWaveHeight );
+    bool shouldIGoOut( int fatigueLevel );
+    float chooseBoardSize();
+};
+
+SurfReport::SurfReport()
+{
+    waveHeight = 3.5f;
+    wavePeriod = 18.f;
+    windDirection = "SSE";
+    windSpeed = 10;
+    std::cout<< "a SurfReport was created" << std::endl;
+}
+
+SurfReport::~SurfReport()
+{
+    std::cout<< "a SurfReport was destroyed" << std::endl;
+}
+
+SurfReport::Surfer::Surfer() :
+surferAge(35)
+{
+    std::cout<< "a Surfer aged: " << surferAge << " was created" << std::endl;
+}
+
+SurfReport::Surfer::~Surfer()
+{
+    std::cout<< "a Surfer aged: " << surferAge << " was destroyed" << std::endl;
+}
+
+void SurfReport::Surfer::levelUp( int improvementAmount )
+{
+    if (surferSkill <= 10)
+    {
+        for (int x = 0; x <= improvementAmount && surferSkill <= 10; x++)
+
+            surferSkill += 1; 
+    }
+}
```

Comment:
either remove the blank line in your `for()` loop or add curly `{ }` braces
- [ ] revise